lib/ansible/plugins/httpapi/fortimanager.py

```diff
@@ -130,24 +139,30 @@ def logout(self):
 
     def send_request(self, method, params):
         """
-        Responsible for actual sending of data to the connection httpapi base plugin. Does some formatting as well.
+        Responsible for actual sending of data to the connection httpapi base plugin. Does some formatting too.
         :param params: A formatted dictionary that was returned by self.common_datagram_params()
         before being called here.
         :param method: The preferred API Request method (GET, ADD, POST, etc....)
         :type method: basestring
 
         :return: Dictionary of status, if it logged in or not.
         """
-
         try:
             if self.sid is None and params[0]["url"] != "sys/login/user":
-                raise FMGBaseException("An attempt was made to login with the SID None and URL != login url.")
+                # If not connected, send connection request.
+                if not self.connection._connected:
```

Comment:
You don't need to branch on this, `_connect()` already does the check.

Reply:
10-4